doc/source/Versioning.rst

```diff
@@ -0,0 +1,134 @@
+
+We always encourage users to download the latest commit when installing Grackle.
+Our general policy to keep Grackle's main branch stable and in working order.
+
+Periodically, we "release" new versions of Grackle.
+A release serves 2 purposes:
+
+  1. It's way to designate a development milestone (whether it's new functionality, changes in the public API, bugfixes) at a snapshot in the project's development.
+  2. It's a way to announce the latest changes to the library since the last milestone.
+
+Currently, the repository tracks 2 separate version numbers: the primary version number (used by the documentation and the shared/static library) **and** the pygrackle version number.
+
+.. _primary-version-number:
+
+Primary Version Number
+----------------------
+
+The version number used in a release has the form ``<MAJOR>.<MINOR>(.<MICRO>)``, and the version number tracked in the repository at any given time follows the more general form ``<MAJOR>.<MINOR>(.<MICRO>)(.dev<DEV_NUM>)``.
+In this template ``<MAJOR>``, ``<MINOR>``, and ``<MICRO>`` correspond to major, minor, and micro version numbers (the micro version number may be omitted if it’s zero). The final section can specify a development version.
+
+Version numbers associated with a release follow the rules of `semantic versioning <https://semver.org/>`__. [#f1]_
+
+
+.. COMMENT BLOCK
+
+   If we were following the rules of semantic versioning as strictly
+   as possible, we would NEVER remove deprecated functionality except
+   when incrementing the major version number.  For example, in the
+   release of Grackle 3.0, we marked some functions as deprecated.
+   Technically, if we were adhearing to the rules of semantic
+   versioning as strictly as possible, we would have waited until
+   Grackle 4.0 to actually remove those functions.  The main reason to
+   do this is if we cared about maintaining ABI compatability between
+   minor versions (it would make the shared library numbering make
+   more sense)
+
+   With all of that said, it's still technically correct for us to say
+   that we follow the semantic versioning specification. An explicit
+   exception appears to be carved out for removing deprecated
+   functionality.
+
+A change in the major version number typicially denotes significant changes in functionality.
+It must also be incremented in a new release if a backwards incompatible change has been made to the public API since the last release (alternatively, parts of the public API can be deprecated in a major release and be removed later on in a minor release).
+
+A change in the minor version number can indicate changes in functionality.
+It must be incremented in a new release if a backwards-compatable change has been made to the public API since the last release.
+
+The presence of ``.dev<DEV_NUM>`` in a version number denotes a development version.
```

Comment:
As I mention below, if you'd prefer to switch to `-dev<DEV_NUM>` moving forward, I am ok with that.